### PR TITLE
Siamese Network-48239259-ManandeepSingh

### PR DESCRIPTION
**Pull request by 24msingh24**
This report demonstrates the use of a Siamese Network for Alzheimer's Disease detection using deep learning techniques. The Siamese Network is designed to compare pairs of medical images and determine whether they belong to the same class (e.g., Alzheimer's Disease or Cognitive Normal).
The report is organized into four main components:

1. **modules.py**: Contains the source code for the components of the Siamese Network, including the network architecture and the contrastive loss function.

2. **dataset.py**: Defines the data loader for loading and preprocessing the medical image data.

3. **train.py**: Contains the code for training the Siamese Network, including data loading, model training, and saving the trained model.

4. **predict.py**: Demonstrates the usage of the trained model for image classification, including classifying test images and calculating accuracy.

The trained Siamese Network achieved 

- Test Accuracy: >= 84%

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: OK
- images of results are not displayed in the readme, but I've checked the folder 
- some minimal information is provded, but no figures or specific details
- results are minimal, no discussion/interpretation

Commit messages: OK

Code: 
- only a siamese network is implemented, no classifier
- design is ok
- siamese was trained for only 6 epochs? 
- sufficiently commented
- no experiments/hyperparameter tuning

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification of individual images without a query image**
- siamese network achieves 84% for test set, but no accuracy plots are provided for train or validation
- no classification performance numbers

General comments:
- **PR is in the main branch**
- **algorithm solves the respective problem only half-way**
- you need to have trained a separate classifier that can classify an individual image as AD or NC with their features extracted by the trained siamese network

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Partially Solves problem, no single image classifier -3 
Driver Script present
File structure present, but not mergable because of conflicts -2
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, but broken links -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, fix merge conflicts -2
Request Description good

**Comment by shakes76** [comment]
Please resolve conflicts to allow merge, to recieve feedback marks

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by 24msingh24** [comment]
I sent mail to Shakes for merge conflicts doubt. But i didnt receive any
reply to it.

**Comment by 24msingh24** [comment]
Didn't receive any reply.
